### PR TITLE
Feature fa icons

### PR DESCRIPTION
**Pull request by zhuiks**
Introducing FontAwesome with quite a few icons added: #61

### CODE REVIEW

**Review by tobias-klein — CHANGES_REQUESTED** [review]
Thank you for this, this looks pretty good! The icons fit in very well!

Here are some things that still need to be optimized a bit:

* The icons expand the horizontal space needed in the menu area. The effect is that overall more horizontal screen space is needed. We need to make adjustments in `app/frontend/helper/ui_helper.js` (Method `resizeAppContainer`) to ensure that the "responsive layout" still works when the size of the window is changed to a smaller or bigger size. You can try this by simply resizing the window horizontally and observing the effect on the layout.
* Even with a fairly large horizontal window size the space between the settings icon and the search button gets eliminated. This should not happen.
image
* Previously the menu was wrapping into more lines when the window size was reduced (3-4 lines). This is also managed in the above mentioned method in combination with CSS. Now it looks like below, which is what you would get on a 7" Android Tablet. We need to keep the wrapping into more lines on smaller screens. Can you achieve that? (I suggest to try to keep the wrapping logic as before)
image
* With a small window size the settings menu now gets cut off, previously the button was wrapped all the way to the left side of the window to prevent that effect.
image
* The full screen icon is now vertically slightly bigger than horizontally, which looks a bit odd. Can you format it so that it becomes a square?
image
* Do you think we can enlarge the settings icon? I think it would be nicer if it is slightly bigger to be visually more prominent.
* The acceptance test suite is currently failing. However, this only happens when the window is not big enough (it passes if you maximize the window during the test run), somehow text is cut off and that influences the test result. I'll have another look here and give more feedback later.

File: css/main.css [hunk 1]
```diff
@@ -317,7 +336,8 @@ h2 {
 }
 
 .verse-content {
-  margin-left: 8em; /* Needs to be bigger to support notes and x-refs, aligned with .verse-reference */
+  margin-left: 8em;
+  /* Needs to be bigger to support notes and x-refs, aligned with .verse-reference */
```
[the remark this answers is not in the document]
Comment (zhuiks) [reply css/main.css:340]: Sorry, I had auto-formatter on save. I turned it  off

**Comment by tobias-klein** [comment]
Thanks for the updates, I think all my review points were resolved with your latest changes!
Before merging this to master I would like to give this another test run on Android (via [ezra-project-cordova](https://github.com/ezra-project/ezra-project-cordova.git)).

**Comment by tobias-klein** [comment]
Can we add a little bit of horizontal padding/margin on the left side of the tag/note indicator icons?

Bildschirmfoto 2021-03-18 um 08 27 07

**Comment by tobias-klein** [comment]
image

This is how it looks now on (tested on Linux). I have some more feedback:
* The text loading indicator has moved (you see the two dots in the screenshot, to the right side of the Selection: None label. Previously it was horizontally aligned to the right side of the window and I think that was better. Is it possible to restore the previous formatting for that?
* For the buttons *Change tags*, *New tag* and *Tag statistics* the horizontal space between the icon and the label could be a little bit bigger. I would say this space should be the same for buttons (compare with Book, Tags, Search).

I also tested on Android and things are working without issues.
The only general thing I noticed (also on PC) is that the loading of the icons is slightly delayed after everything else is rendered. I wonder whether this could be changed (but how?!). So currently there is a slight moment where you just see the buttons without icons and then a split second later the icons are shown.

**Comment by zhuiks** [comment]
> The text loading indicator has moved (you see the two dots in the screenshot, to the right side of the Selection: None label. Previously it was horizontally aligned to the right side of the window and I think that was better. Is it possible to restore the previous formatting for that?

Yes, it's possible. I add that

> For the buttons _Change tags_, _New tag_, _Tag statistics_, _Compare_, _Context_ the horizontal space between the icon and the label could be a little bit bigger. I would say this space should be the same for all buttons (compare with Book, Tags, Search).

For buttons _book_, _Tags_ and _Search_ the constant width is set. Do you want those buttons to different width? (i.e _Search_ will be bigger because it's a longer word) 
For all other buttons their width depends on the content. I just had separated icons with `&nbsp;`  I'll replace it with span. Then the space can be adjusted with css.

Also please notice that for some icons it's actually two icons.

> The only general thing I noticed (also on PC) is that the loading of the icons is slightly delayed after everything else is rendered. I wonder whether this could be changed (but how?!). So currently there is a slight moment where you just see the buttons without icons and then a split second later the icons are shown.

I think it takes time to get Font-awesome webfont. I'll see what are the options. Maybe font-awesome is not the best... Material and Bootstrap having their own icon sets now or there are [pure css icons](https://css.gg/app). So we can switch in the process of big refactoring under #196 And leave font-awesome for now. What do you think?

**Comment by zhuiks** [comment]
I think I got everything including fa webfont :)

**Comment by tobias-klein** [comment]
The preloading of fa-webfont worked! That's better!
But now ... for some reason the menu panel always has four lines ... independent of the available window width. Did something change there?
image

Regarding the button labels, something general that we need to consider is the texts may be smaller or bigger based on the locale. For example French texts are generally longer than English or German ones.

**Comment by zhuiks** [comment]
> But now ... for some reason the menu panel always has four lines ... independent of the available window width. Did something change there?

Yes, I did right aligning  of the loader incorrectly. 
It's fixed now!

> Regarding the button labels, something general that we need to consider is the texts may be smaller or bigger based on the locale. For example French texts are generally longer than English or German ones.

Yes, all the buttons have dynamic width now. Here is example with longer Russian words:
ezra-project_index html(iPad)

**Review by tobias-klein — APPROVED** [review]

**Review by tobias-klein — APPROVED** [review]

**Comment by tobias-klein** [comment]
Thanks once more! Good job :)

**Pull request merged**